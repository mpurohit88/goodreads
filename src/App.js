import React, { Component } from 'react';
import './App.css';
import Tile from './BookTile';
import { get } from './httpClient';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {}
  }

  handleKeyPress = e => {
    if(e.charCode === 13){
      this.setState({isLoading: true});
      get(e.target.value, this.callback).then(searchResult => this.setState({ posts: searchResult, isLoading: false}));
    }
  }

  render() {
    const result = this.state.posts ? this.state.posts.GoodreadsResponse.search.results : undefined;

    return [
        <div className="App">
          <InputLabel htmlFor="search">Provide Book Title and Press Enter (ex. react)</InputLabel>&nbsp;&nbsp;&nbsp;&nbsp;
          <Input type="text" name="search" id="search" onKeyPress={this.handleKeyPress} autoFocus />
          {
            this.state.isLoading && <span id="loading">Loading...</span>
          }
          {
            result === null ? <span id="notFound">No Result Found</span> : "" 
          }
          {
            result && Array.isArray(result.work) ? result.work.map(function(searchResult, index) {
              return <Tile searchResult={searchResult} index={index} />
            }) : result && <Tile searchResult={result.work} index={1} />
          }
        </div>
    ]
  }
}

export default App;
