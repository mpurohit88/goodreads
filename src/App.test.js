import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, configure, mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('render home page correctly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();

});

it('should be possible to activate button with Spacebar', () => {
  const component = shallow(<App />);

  component
    .find('WithStyles(Input)#search')
    .simulate('keypress', { keyCode: 13, target: {value: "react"} });
  expect(component).toMatchSnapshot();
  component.unmount();
});
