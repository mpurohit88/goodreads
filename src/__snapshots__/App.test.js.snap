// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render home page correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(InputLabel)
          htmlFor="search"
        >
          Provide Book Title and Press Enter (ex. react)
        </WithStyles(InputLabel)>,
        "    ",
        <WithStyles(Input)
          autoFocus={true}
          id="search"
          name="search"
          onKeyPress={[Function]}
          type="text"
        />,
        undefined,
        "",
        undefined,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Provide Book Title and Press Enter (ex. react)",
          "htmlFor": "search",
        },
        "ref": null,
        "rendered": "Provide Book Title and Press Enter (ex. react)",
        "type": [Function],
      },
      "    ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "id": "search",
          "name": "search",
          "onKeyPress": [Function],
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
      "",
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(InputLabel)
            htmlFor="search"
          >
            Provide Book Title and Press Enter (ex. react)
          </WithStyles(InputLabel)>,
          "    ",
          <WithStyles(Input)
            autoFocus={true}
            id="search"
            name="search"
            onKeyPress={[Function]}
            type="text"
          />,
          undefined,
          "",
          undefined,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Provide Book Title and Press Enter (ex. react)",
            "htmlFor": "search",
          },
          "ref": null,
          "rendered": "Provide Book Title and Press Enter (ex. react)",
          "type": [Function],
        },
        "    ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": true,
            "id": "search",
            "name": "search",
            "onKeyPress": [Function],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
        "",
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`should be possible to activate button with Spacebar 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(InputLabel)
          htmlFor="search"
        >
          Provide Book Title and Press Enter (ex. react)
        </WithStyles(InputLabel)>,
        "    ",
        <WithStyles(Input)
          autoFocus={true}
          id="search"
          name="search"
          onKeyPress={[Function]}
          type="text"
        />,
        undefined,
        "",
        undefined,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Provide Book Title and Press Enter (ex. react)",
          "htmlFor": "search",
        },
        "ref": null,
        "rendered": "Provide Book Title and Press Enter (ex. react)",
        "type": [Function],
      },
      "    ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "id": "search",
          "name": "search",
          "onKeyPress": [Function],
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
      "",
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(InputLabel)
            htmlFor="search"
          >
            Provide Book Title and Press Enter (ex. react)
          </WithStyles(InputLabel)>,
          "    ",
          <WithStyles(Input)
            autoFocus={true}
            id="search"
            name="search"
            onKeyPress={[Function]}
            type="text"
          />,
          undefined,
          "",
          undefined,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Provide Book Title and Press Enter (ex. react)",
            "htmlFor": "search",
          },
          "ref": null,
          "rendered": "Provide Book Title and Press Enter (ex. react)",
          "type": [Function],
        },
        "    ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": true,
            "id": "search",
            "name": "search",
            "onKeyPress": [Function],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
        "",
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
